private/pkg/manifest/module.go

```diff
@@ -169,7 +170,8 @@ func NewBlobSet(ctx context.Context, blobs []Blob, opts ...BlobSetOption) (*Blob
 	}
 	digestToBlobs := make(map[string]Blob, len(blobs))
 	for _, b := range blobs {
-		digestStr := b.Digest().String()
+		digest := b.Digest()
+		digestStr := digest.String()
```

Comment:
From downside 1: this sometimes could be a bit verbose

Reply:
Digests has pointer receivers. If we really wanted to return Digest as values, I think we should be considering Digest values as immutable and define only value receivers. Mixing the two is causing headaches like these.